source/Posets/InductiveTypesfromSmallGeneratedLattice.lagda

```diff
@@ -0,0 +1,1768 @@
+Ian Ray 01/09/2023.
+
+We formalize Curi's notion of Abstract Inductive Definition (CZF) within the
+context of a sup-lattice L with small basis B (and q : B → L). An abstract
+inductive defintion is a subset ϕ : B × L → Prop which can be thought of as a
+'inference rule' concluding b from a. An inductive definition induces a
+closure condition. More precisely, a subset S is closed under ϕ if for all
+b : B and a : L such that (b , a) ∈ ϕ and ↓ᴮa is 'contained' in S then b ∈ S.
+Interestingly, there is an intimate connection between the least-closed
+subsets of some inductive definition ϕ and the least fixed point of a monotome
+map related to ϕ in a precise way. In this file we will develop this
+relationship and prove a predicative version of the least fixed point theorem.
+This work follows the paper 'On Tarski's Fixed Point Theorem' by Giovanni Curi.
+Fortunately, unlike in the realm of set theory, induction rules are first
+class citizens in type theory. Using UF + HITs we can construct the least
+closed subset, under an inductive definition ϕ, as a special Quotient Inductive
+Type (QIT). 
+
```

Comment:
Please include a full bibliographic reference to Curi's work here, including, if available, a publicly available version.

Reply:
Can you also formulate, informally but rigorously, in mathematical vernacular, the main theorem(s) proved in this file, at this point?

Reply:
Yes. I will add those now. I am assuming you want the result stated at the beggining of the file.